api/service/legacysync/syncing.go

```diff
@@ -1079,9 +1083,9 @@ func (ss *StateSync) SyncLoop(bc core.BlockChain, worker *worker.Worker, isBeaco
 
 	for {
 		start := time.Now()
-		otherHeight := getMaxPeerHeight(ss.syncConfig)
 		currentHeight := bc.CurrentBlock().NumberU64()
-		if currentHeight >= otherHeight {
+		otherHeight, errMaxHeight := getMaxPeerHeight(ss.syncConfig)
+		if errMaxHeight != nil || currentHeight >= otherHeight {
```

Comment:
Add a separate log here if `errMaxHeight != nil` so that we can easily `grep` the occurrence of a syncing failure. It happens if the node forks, or if the DNS is pointed to nodes with snap database.

Reply:
added